driver-core/src/main/com/mongodb/internal/async/client/AsyncMapReduceIterable.java

```diff
@@ -138,7 +138,9 @@
      * @param sharded if the output database is sharded
      * @return this
      * @mongodb.driver.manual reference/command/mapReduce/#output-to-a-collection-with-an-action output with an action
+     * @deprecated this option will no longer be supported in MongoDB 4.4
      */
+    @Deprecated
```

Comment:
I'm still not clear why there are deprecations for internal classes.  It seems harmless but also unnecessary.

Reply:
ACK - apologies didnt engage brain - done.